src/events.c

```diff
@@ -90,22 +90,51 @@ static struct events_node *new_events_node(struct event_t *event) {
 }
 
 void install_event(struct events_list *events, struct event_t *event) {
+	struct events_node *new_node = new_events_node(event);
+
 	if (events->head == NULL) {
 		// First event for this control
-		struct events_node *new_node = new_events_node(event);
 		events->head = new_node;
 		events->tail = new_node;
 		return;
 	}
 
-	// TODO: we need to remove existing events_node for the same event
-	// and support NULL event to only remove event.
+	if (strcmp(event->name, events->head->event->name) == 0) {
+		// clear_event(events->head->event);
```

Comment:
Uncommenting this leads to a crash (with the tests):
```
....../libui-napi/tests/run-internals.js:14
	check_c_tests.run_tests();
	              ^

Error: napi_reference_unref failed with code 0: Unknown failure
```

Reply:
Was there a reason for this 0 to `napi_create_reference`:

```c
static struct event_t *make_event(napi_env env, char *name) {
	napi_value object;
	napi_create_object(env, &object);
	napi_ref ref;
	napi_create_reference(env, object, 0, &ref);
	struct event_t *event = create_event(env, ref, name);
	return event;
}
```

Reply:
I'll check it tomorrow, I'm too tired and I have to rest 😴.
BTW, I don't think we need to use strcmp, since we only use string literals as test names (AFAIK), the two string should be two equal pointers...

Reply:
> Is there a reason for passing 0 to napi_create_reference? (tests install_event.c)

No, setting it to 1 at creation seems to solve the problem